### PR TITLE
add Xu-Needleman cohesive material

### PR DESCRIPTION
**Pull request by kimauth**
Finally started the cohesive materials.

### CODE REVIEW

File: test/test_xu_needleman.jl [hunk 1]
```diff
@@ -0,0 +1,19 @@
+@testset "XuNeedleman" begin
+    m = MaterialModels.XuNeedleman(σₘₐₓ = 400., τₘₐₓ=200., δₙ=0.002, δₜ=0.002, Δₙˢ=0.01)
+
+    # 3D
+    Δ = Tensor{1,3}((0.0, 0.0, m.δₙ))
+    T, dTdΔ, state = material_response(m, Δ)
+    @test T[end] ≈ m.σₘₐₓ
+    Δ = Tensor{1,3}((sqrt(2)*m.δₜ/2, 0.0, 0.0))
+    T, dTdΔ, state = material_response(m, Δ)
+    @test T[1] ≈ m.τₘₐₓ
```
Comment (lijas) [inline test/test_xu_needleman.jl:10]: Call the check_jld2 functions aswell? (Maybe not worth it for this small material model)

File: src/CohesiveMaterials/XuNeedleman.jl [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+    if !p_strength 
+        σₘₐₓ = Φₙ / (exp(1.0) * δₙ)
+        τₘₐₓ = Φₜ / (sqrt(0.5exp(1.0)) * δₜ)
+    elseif !p_jump
+        δₙ = Φₙ / (exp(1.0) * σₘₐₓ)
+        δₜ = Φₜ / (sqrt(0.5exp(1.0)) * τₘₐₓ)
+    elseif !p_work
+        Φₙ = σₘₐₓ * exp(1.0) * δₙ
+        Φₜ = τₘₐₓ * sqrt(0.5exp(1.0)) * δₜ
```
[the remark this answers is not in the document]
Comment (kimauth) [reply src/CohesiveMaterials/XuNeedleman.jl:54]: They come straight from the paper, let's assume Xu and Needleman knew what they were writing about :P

File: src/CohesiveMaterials/XuNeedleman.jl [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    σₘₐₓ::Float64
+    τₘₐₓ::Float64
+    δₙ::Float64
+    δₜ::Float64
+    Φₙ::Float64
+    Φₜ::Float64
+    Δₙˢ::Float64
+end
+
+function XuNeedleman(;
```
[the remark this answers is not in the document]
Comment (kimauth) [reply src/CohesiveMaterials/XuNeedleman.jl:32]: Yeah, I am not super happy with the syntax either. We could supply helper functions like `Φₙ(σₘₐₓ, δₙ)`. Quite clutters the namespace though if we supply all 6 of them.

File: src/CohesiveMaterials/XuNeedleman.jl [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+- `δₙ::Float64`: characteristic normal separation
+- `δₜ`::Float64`: characteristic tangential separation
+- `Φₙ::Float64`: normal work of separation
+- `Φₜ::Float64`: tangential work of separation
+- `Δₙˢ`: normal separation after complete shear separation under the condition of zero normal tension
+
+Out of the pairs cohesive strength / characteristic normal separation / work of separation any two can be given
+for construction the `XuNeedleman` material.
+"""
+struct XuNeedleman <: AbstractMaterial
```
[the remark this answers is not in the document]
Comment (kimauth) [reply src/CohesiveMaterials/XuNeedleman.jl:22]: Yeah, maybe. I didn't because there was no usage of it for now.

File: test/test_xu_needleman.jl (hunk 1 above)
Comment (kimauth) [reply test/test_xu_needleman.jl:10]: Yeah, I think that's more important for the ones which have state variables. There could be more tests though, e.g. for mixed loading.

File: src/CohesiveMaterials/XuNeedleman.jl (hunk 4 above)
[the remark this answers is not in the document]
Comment (kimauth) [reply src/CohesiveMaterials/XuNeedleman.jl:22]: Let's leave it alone until we find a place where it would be actually useful.

File: src/CohesiveMaterials/XuNeedleman.jl [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+    # unpack variables
+    Φₜ = m.Φₜ
+    Φₙ = m.Φₙ
+    δₜ = m.δₜ
+    δₙ = m.δₙ
+    q = Φₜ/Φₙ
+    r = m.Δₙˢ / δₙ
+
+    Φ(Δₜ, Δₙ) = Φₙ + Φₙ*exp(-Δₙ/δₙ) * ((1 - r + Δₙ/δₙ) * (1-q)/(r-1) - (q + (r-q)/(r-1) * Δₙ/δₙ) * exp(-(Δₜ ⋅ Δₜ) / δₜ^2))
```
Comment (fredrikekre) [inline src/CohesiveMaterials/XuNeedleman.jl:84]: Drive-by comment, but in the paper they have `Φ(Δn, Δt)` instead of `Φ(Δₜ, Δₙ)` IIUC.
